test/syntax.jl

```diff
@@ -2751,3 +2751,8 @@ end
 @test eval(:(x = $(QuoteNode(Core.SlotNumber(1))))) == Core.SlotNumber(1)
 @test_throws ErrorException("syntax: SSAValue objects should not occur in an AST") eval(:(x = $(Core.SSAValue(1))))
 @test_throws ErrorException("syntax: Slot objects should not occur in an AST") eval(:(x = $(Core.SlotNumber(1))))
+
+# issue 40258
+@test "a $("b $("c")")" == "a b c"
+
+@test "$(([[:a, :b], [:c, :d]]...)...)" == "abcd"
```

Comment:
It might be worth also testing that we still support both forms of the AST, in case anyone was manually constructing ASTs in the old form?:

```julia
julia> VERSION
v"1.5.4"

julia> eval(Expr(:string, "a", Expr(:string, "b", Expr(:string, "c"))))
"abc"

julia> eval(Expr(:string, "a", Expr(:string, "b", "c")))
"abc"
```

```julia
julia> VERSION
v"1.6.0"

julia> eval(Expr(:string, "a", Expr(:string, "b", Expr(:string, "c"))))
"ab"

julia> eval(Expr(:string, "a", Expr(:string, "b", "c")))
"ab"
```

Reply:
So something like
```suggestion
@test "$(([[:a, :b], [:c, :d]]...)...)" == "abcd"

@test eval(Expr(:string, "a", Expr(:string, "b", "c"))) == "abc"
@test eval(Expr(:string, "a", Expr(:string, "b", Expr(:string, "c")))) == "abc"
```

Reply:
Thanks! Always good to have more test cases.